languages/python/sqlalchemy-oso/tests/test_roles2.py

```diff
@@ -299,6 +346,8 @@ def test_resource_with_roles_no_actions(init_oso, sample_data):
     oso.roles.assign_role(leina, osohq, "member", session)
     oso.roles.assign_role(steve, oso_repo, "repo_read", session)
 
+    session.commit()
```

Comment:
Why are all these `session.commit()` calls suddenly required?

Reply:
Because we change the roles with one session and then the is_allowed queries use new sessions. I think it makes sense that these have to be committed, I don't really know why it works on sqlite without committing them, maybe just some odd behavior of the in memory sqlite.

Reply:
We could just not pass sessions to the `assign_role` calls and that would work too.

Reply:
> We could just not pass sessions to the `assign_role` calls and that would work too.

up 2 u

Reply:
> then the is_allowed queries use new sessions.

ah not that `is_allowed()` uses a session but that it's operating over uncommitted data, right? Wonder if `.flush()` would do the trick here, too.

> I don't really know why it works on sqlite without committing them, maybe just some odd behavior of the in memory sqlite.

yeah this seems real suspect to me — I'd bet it's some autocommit-on-access behavior, but that seems like it'd be a SQLAlchemy-level configuration instead of database specific.